docs/general.md

```diff
@@ -236,8 +236,58 @@ There are three types of `far_call`:
 
 ## Precompiles and System calls
 
-Explain how calls to precompiles work, use keccak as an example as it's used on every deployment (it goes to the `keccak.yul` contract which then uses the `precompile` opcode).
-What are system contracts? What's a system call? Show some examples (deployer, nonce holder, L2BaseToken) and what they're used for.
+### Precompiles
+
+Precompiles are special, highly optimized functions embedded within the EraVM. Precompiles exist natively within the EraVM and are designed to handle computationally intensive operations efficiently.
+
+When a contract calls a precompile, it does so by invoking a specific address range that the EraVM recognizes as a precompile. These addresses are hardcoded and correspond to specific operations. For instance, the `keccak256` hash function, is implemented as a precompile.
+
+> [!NOTE]
+> The `keccak256` hash function, commonly used for generating unique identifiers and cryptographic proofs, is used in almost every contract deployment.
+
+#### Example: `keccak256` Precompile
+
+Here's a step-by-step overview of how it works after invoking the `Precompile` opcode:
+
+1. **Identifying the Precompile Address**: The address used for the `keccak256` execution is derived from the last two bytes of the current contract's address (`address_low`). If this matches the predefined `keccak256` address, the system directs the execution to the appropriate function.
+    ```rust
+    let address_bytes = vm.current_context()?.contract_address.0;
+    let address_low = u16::from_le_bytes([address_bytes[19], address_bytes[18]]);
+    if address_low == KECCAK256_ROUND_FUNCTION_PRECOMPILE_ADDRESS {
+        keccak256_rounds_function(abi_key, heaps)?;
+    }
+    ```
+
+2. **Execution in `keccak.yul`**: The `keccak256_rounds_function` processes the input data in blocks, applies the _Keccak hash algorithm_, and stores the intermediate hash state in the contract's memory, rather than the finalized digest.
```

Comment:
I think it is not sufficiently clear that when someone needs to execute a precompile it calls a certain system contract, and that is the one that executes the opcode, in this case keccak.yul is the one executing the precompile opcode